pkg/workload/workload_generator/ddl_generator.go

```diff
@@ -0,0 +1,368 @@
+package workload_generator
+
+import (
+	"bufio"
+	"encoding/csv"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/cockroachdb/errors"
+)
+
+// ParseDDL converts a "CREATE TABLE ..." DDL statement into a TableSchema.
+// It parses the table name, columns, and constraints (primary keys, unique constraints,
+// foreign keys, and check constraints) from the DDL statement and returns a structured
+// representation of the table schema.
+func ParseDDL(ddl string) (*TableSchema, error) {
+	// Define regex patterns for identifying identifiers in SQL
+	ident := `(?:"[^"]+"|[A-Za-z_][\w]*)`                       // Matches quoted or unquoted identifiers
+	fullIdent := fmt.Sprintf(`(%s(?:\.%s){0,2})`, ident, ident) // Matches fully qualified names (up to 3 parts)
+
+	// Create regex to match CREATE TABLE statements and extract the table name
+	tablePattern := regexp.MustCompile(`(?i)CREATE\s+TABLE(?:\s+IF\s+NOT\s+EXISTS)?\s+` + fullIdent)
+	m := tablePattern.FindStringSubmatch(ddl)
+	if m == nil {
+		return nil, errors.New("no table name")
+	}
+	// Extract and normalize the table name
+	tableName := m[1]
+	parts := strings.Split(tableName, ".")
+	for i := range parts {
+		parts[i] = strings.Trim(parts[i], `"`) // Remove quotes from parts
+	}
+	rawName := tableName                 // Save the original table name
+	tableName = strings.Join(parts, ".") // Reconstruct normalized table name
+
+	// Create a new TableSchema with the normalized name
+	ts := NewTableSchema(tableName, rawName)
+
+	// Extract the column definitions block from the DDL
+	bodyRe := regexp.MustCompile(`(?s)\((.*)\)\s*([^)]*)$`)
+	bodyMatch := bodyRe.FindStringSubmatch(ddl)
+	if bodyMatch == nil {
+		return nil, errors.New("no column block")
+	}
+	body := bodyMatch[1]
+
+	var partsList []string
```

Comment:
from here till we populate colDefs, tableConstraints - this should be a function that return colDefs, tableConstraints.